lib/livebook/intellisense/docs.ex

```diff
@@ -95,6 +106,7 @@ defmodule Livebook.Intellisense.Docs do
               documentation: documentation(doc, format),
               signatures: signatures,
               specs: Map.get(specs, {name, base_arity}, []),
+              type_spec: Map.get(type_specs, {name, base_arity}, nil),
```

Comment:
It's always either of the to, so let's unify by putting the type spec in specs as well. We can build the specs map with keys `{kind, name, base_arity}`.

Reply:
Originally, I did return the type spec on the `specs` field as well. There's something about it that does not sit quite right, maybe you have some more thoughts on it, 

For types, there's only ever one spec, so `specs` would always be either `[]` or `[type_spec]`, matching on that downstream instead of `nil` / `type` feels dirty, but maybe that's just me?

Reply:
As for collecting function / type specs into one map, is there a better way to do it than:
```elixir
specs =
  with true <- :function in kinds or :macro in kinds,
       {:ok, specs} <- Code.Typespec.fetch_specs(module) do
    for {{name, arity}, spec} <- specs,
        into: Map.new(),
        do: {{:function, name, arity}, spec}
  else
    _ -> %{}
  end

specs =
  with true <- :type in kinds,
       {:ok, types} <- Code.Typespec.fetch_types(module) do
    for {type_kind, {name, _defs, vars}} = type <- types,
        type_kind in [:type, :opaque],
        into: specs,
        do: {{:type, name, Enum.count(vars)}, [type]}
  else
    _ -> specs
  end
```
?

Reply:
I think `[]` and `[type_spec]` makes sense, since we have the same map structure for all module members (functions, macros and types) and a list of specs is generic to accommodate for all cases. We could have separate maps for each kind, but I'd rather keep it the same for simplicity :)

Reply:
Actually, I changed my mind, we can keep `:type_spec`, just please update `@type member_info` at the top of the module, thanks!

Reply:
👍 Of course I missed extending a type definition on a PR about types...